packages/api/models/granules.js

```diff
@@ -29,6 +33,27 @@ class Granule extends Manager {
     super(process.env.GranulesTable, granuleSchema);
   }
 
+  /**
+  * Adds fileSize values from S3 object metadata for granules missing that information
+  *
+  * @param {Array<Object>} files - Array of files from a payload granule object
+  * @returns {Promise<Array>} - Updated array of files with missing fileSize appended
+  */
+  addMissingFileSizes(files) {
+    const filePromises = files.map((file) => {
+      if (!('fileSize' in file)) {
+        const filePath = typeof file.path === 'undefined' ? '' : `${file.path}/`;
```

Comment:
I'm confused about what this check is for

Reply:
@laurenfrederick there's at least one test fixture that implies it's possible to have a file object with a bucket without a path but a name when the file is in the root of the bucket - this is defensive to avoid a bad result if that occurs.      Alternately we could assume the file.fileName will always be there and pull the bucket/key from that....

Reply:
Got it, thanks!